### ISSUE TITLE
Inaccurate event type definition

### ISSUE THREAD

**Issue by nightire** [cibernox/ember-basic-dropdown#552]
The following code claims that the `event` object is a `MouseEvent`, but then this function is also used for `touchend` event:

https://github.com/cibernox/ember-basic-dropdown/blob/ea50fb854475d0e8d865ab8174c5c4138d152e25/addon/components/basic-dropdown-content.ts#L90

So, it may be better to be more precise as:

```typescript
this.handleRootMouseDown = (e: MouseEvent | TouchEvent): any => { ... }
```

or simply more general as:

```typescript
this.handleRootMouseDown = (e: Event): any => { ... }
```

Note, this issue only affects users are:

1. using typescript
2. and extending class(es) from ECD
3. compiling `.ts` code or generating type definitions

**Comment by nightire** [comment]
@cibernox May you take a look at this issue? It breaks the publishing of our own add-ons because of extending the components from `ember-basic-dropdown`.

**Comment by cibernox** [comment]
Fixed and released in 3.0.4! Thanks for pointing this out.
